crates/ethrpc/src/alloy/mod.rs

```diff
@@ -23,3 +26,28 @@ pub fn provider(url: &str) -> AlloyProvider {
         .http(url.parse().unwrap());
     ProviderBuilder::new().connect_client(rpc).erased()
 }
+
+pub fn provider_with_account(url: &str, private_key: &PrivateKey) -> anyhow::Result<AlloyProvider> {
+    let rpc = ClientBuilder::default()
+        .layer(LabelingLayer {
+            label: "main".into(),
+        })
+        .layer(InstrumentationLayer)
+        .layer(BatchCallLayer::new(Default::default()))
```

Comment:
If this function is intended to do be used to issues just a handful of transactions maybe we should avoid `BatchCallLayer` as that spawns a separate task for every instance.